client/components/filters/filter/index.js

```diff
@@ -207,31 +211,52 @@ FilterPicker.propTypes = {
 	/**
 	 * An array of filters and subFilters to construct the menu.
 	 */
-	filters: PropTypes.arrayOf(
-		PropTypes.shape( {
-			/**
-			 * A custom component used instead of a button, might have special handling for filtering. TBD, not yet implemented.
-			 */
-			component: PropTypes.string,
-			/**
-			 * The label for this filter. Optional only for custom component filters.
-			 */
-			label: PropTypes.string,
-			/**
-			 * An array representing the "path" to this filter, if nested.
-			 */
-			path: PropTypes.string,
-			/**
-			 * An array of more filter objects that act as "children" to this item.
-			 * This set of filters is shown if the parent filter is clicked.
-			 */
-			subFilters: PropTypes.array,
-			/**
-			 * The value for this filter, used to set the `filter` query param when clicked, if there are no `subFilters`.
-			 */
-			value: PropTypes.string.isRequired,
-		} )
-	).isRequired,
+	config: PropTypes.shape( {
+		/**
+		 * A label above the filter selector.
+		 */
+		label: PropTypes.string,
+		/**
+		 * Url parameters to persist when selecting a new filter.
+		 */
+		staticParams: PropTypes.array.isRequired,
+		/**
+		 * The url paramter this filter will modify.
+		 */
+		param: PropTypes.string.isRequired,
+		/**
+		 * Determine if the filter should be shown. Supply a function with the query object as an argument returning a boolean.
+		 */
+		showFilters: PropTypes.func.isRequired,
+		/**
+		 * An array of filter a user can select.
+		 */
+		filters: PropTypes.arrayOf(
+			PropTypes.shape( {
+				/**
+				 * A custom component used instead of a button, might have special handling for filtering. TBD, not yet implemented.
+				 */
+				component: PropTypes.string,
```

Comment:
If this would be a component, the proptype wouldn't be string correct? Or is this just a temporary since it has yet to be developed?